application/src/main/java/uk/nhs/hee/tis/revalidation/service/GmcDoctorConnectionSyncService.java

```diff
@@ -69,20 +72,50 @@ public void receiveMessage(final String gmcSyncStart) {
     log.info("Message from integration service to start gmc sync {}", gmcSyncStart);
 
     if (gmcSyncStart != null && gmcSyncStart.equals("gmcSyncStart")) {
-      ListUtils.partition(doctorsForDBRepository.findAll(), BATCH_SIZE).stream()
-          .map(batch ->
-              batch.stream().map(this::convertToMessage).filter(Objects::nonNull).toList())
-          .filter(not(List::isEmpty))
-          .forEach(batch -> sqsClient.sendMessageBatch(sqsEndPoint, batch));
-      log.info("GMC doctors have been published to the SQS queue ");
-      final var syncEnd = IndexSyncMessage.builder()
-          .syncEnd(true)
-          .build();
-      try {
-        sqsClient.sendMessage(sqsEndPoint, objectMapper.writeValueAsString(syncEnd));
-      } catch (JsonProcessingException e) {
-        log.error("Unable to convert 'syncEnd' message.  Downstream services need notification.");
-      }
+      List<List<SendMessageBatchRequestEntry>> batches = ListUtils.partition(
+              doctorsForDBRepository.findAll(), BATCH_SIZE).stream()
+          .map(
+              batch -> batch.stream().map(this::convertToMessage).filter(Objects::nonNull).toList())
+          .filter(not(List::isEmpty)).toList();
+
+      List<CompletableFuture<SendMessageBatchResponse>> futures = batches.stream()
+          .map(batch -> sqsClient.sendMessageBatch(
+                  SendMessageBatchRequest.builder().queueUrl(sqsEndPoint).entries(batch).build())
+              .handle((resp, ex) -> { // let the resulting future always complete successfully
+                if (ex != null) {
+                  List<String> ids = batch.stream().map(SendMessageBatchRequestEntry::id).toList();
+                  log.error("Failed to send batch for doctors: {}", ids, ex);
+                } else {
+                  resp.failed()
+                      .forEach(r -> log.error("Failed to send message for doctor {}", r.id()));
+                }
+                return resp;
+              })
+          ).toList();
+
+      CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]))
+          .thenCompose(v -> {
+            log.info("GMC doctors have been published to the SQS queue ");
+            final var syncEnd = IndexSyncMessage.builder().syncEnd(true).build();
+            try {
+              String body = objectMapper.writeValueAsString(syncEnd);
+              return sqsClient.sendMessage(
```

Comment:
is it not possible to continue to use sendMessageBatch here? ir does this achieve the same thing?

Reply:
The syncEnd message is just a single message and it's unnecessary to use `sendMessageBatch` here. In the previous code, we also used `sendMessage`.